components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/auth/OAuthHandler.java

```diff
@@ -0,0 +1,214 @@
+    /* properties map to be initialized */
+    private Map<String, String> properties;
+    /* properties specific to this authenticator */
+    private String remoteServiceURL;
+    private int priority;
+    private String userName;
+    private String password;
+
+    // Ideally this should be configurable. For the moment, hard code the priority.
+
+    public int getPriority() {
+        return priority;
+    }
+
+    public void setPriority(int priority) {
+        this.priority = priority;
+    }
+
+    public void setDefaultPriority() {
+        this.priority = DEFAULT_PRIORITY;
+    }
+
+    public void setDefaultAuthzServer() {
+        this.remoteServiceURL = LOCAL_AUTH_SERVER;
+    }
+
+    public boolean canHandle(Message message, ClassResourceInfo classResourceInfo) {
+        // check the "Authorization" header and if "Bearer" is there, can be handled.
+
+        // get the map of protocol headers
+        Map protocolHeaders = (TreeMap) message.get(Message.PROTOCOL_HEADERS);
+        // get the value for Authorization Header
+        List authzHeaders = (ArrayList) protocolHeaders
+                .get(EntitlementEndpointConstants.AUTHORIZATION_HEADER);
+        if (authzHeaders != null) {
+            // get the authorization header value, if provided
+            String authzHeader = (String) authzHeaders.get(0);
+            if (authzHeader != null && authzHeader.contains(BEARER_AUTH_HEADER)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isAuthenticated(Message message, ClassResourceInfo classResourceInfo) {
+        // get the map of protocol headers
+        Map protocolHeaders = (TreeMap) message.get(Message.PROTOCOL_HEADERS);
+        // get the value for Authorization Header
+        List authzHeaders = (ArrayList) protocolHeaders
+                .get(EntitlementEndpointConstants.AUTHORIZATION_HEADER);
+        if (authzHeaders != null) {
+            // get the authorization header value, if provided
+            String authzHeader = (String) authzHeaders.get(0);
+
+            // extract access token
+            String accessToken = authzHeader.trim().substring(7).trim();
+            // validate access token
+            try {
+                OAuth2ClientApplicationDTO validationApp = this.validateAccessToken(accessToken);
+                OAuth2TokenValidationResponseDTO validationResponse = null;
+
+                if (validationApp != null) {
+                    validationResponse = validationApp.getAccessTokenValidationResponse();
+                }
+
+                if (validationResponse != null && validationResponse.isValid()) {
+                    String userName = validationResponse.getAuthorizedUser();
+                    authzHeaders.set(0, userName);
+
+                    // setup thread local variable to be consumed by the provisioning framework.
+                    RealmService realmService = (RealmService) PrivilegedCarbonContext
```

Comment:
Is this used somewhere? This component should not be setting anything specifically to be consumed by provisioning framework.